webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/UserSessionInvalidationFilter.scala

```diff
@@ -132,3 +138,136 @@ class UserSessionInvalidationFilter(userRepository: UserRepository, userDetailLi
   override protected def shouldNotFilterErrorDispatch = false
 
 }
+
+private[liftweb] object UserSessionInvalidationFilter {
+
+  // Avoid copy method because changing state may cause invalidation
+  sealed abstract class LoggedUser(val status: UserStatus, val password: String) {
+    def withStatus(newStatus:     UserStatus): LoggedUser
```

Comment:
You should not keep a clear form of the password in ram, it's too easy to get access to it (for some value of easy, but let say it's a good practice to not hold that on ram)

Reply:
(saying that, I think we already have it in memory for the user details).

Reply:
Yes we already had it for the User account also. What we could do here is to make a value class, and also override the toString method just in case it is logged

Reply:
=> https://github.com/Normation/rudder/pull/5866/commits/ddd30387f42ccc96481710b0cf6e52c723390097

Reply:
Why do we keep it for User account ? Once authenticated, we don't care anymore. 
But ok, other subject. At least, we should not add more of them at all everywhere.